react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,14 +109,40 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => void;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _onFinalizeServerURL() {
+        const { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(http(?:s)?://)?.+$');
```

Comment:
I think building a URL object provides stronger validation.  If that fails, you could concat `https://` and try again. Then check if the protocol `startsWith` `http`.

Reply:
That seems to be two steps for me and two URL object instantiation. I like it better with regexp and a single step. But thanks for the last point, that makes sense.

Reply:
The regex won't catch all invalid URLs. IMHO, we should be correct, otherwise joining new conferences will fail. In addition, this is not a hot path, so using 2 steps instead of one is moot.

Reply:
Added URL instantiation, but kept the regexp part, because it's easier to reconstruct the URL with that data if need be (see line 132, instead of url.host + url.path + ...). Please also note that this doesn't really do anything ATM as URL is polyfilled on mobile and is very lenient on parsing. Talked to Lyubo, this is a global issue that will be solved eventually, so we agreed that I leave it like this for now. WDYT?